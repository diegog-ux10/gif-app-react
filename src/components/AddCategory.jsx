import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputvalue, setInputvalue] = useState('');

    const onInputChange = (event) => {
        setInputvalue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const valueFix = inputvalue.trim();
        if(valueFix.length <= 1) return;
        onNewCategory(valueFix);
        setInputvalue('');
    }

  return (
    <form onSubmit={(onSubmit)} aria-label="form">

        <input
            type="text"
            value={ inputvalue }
            placeholder="Buscar Gif"
            onChange={ onInputChange } 
        />

    </form>
  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
