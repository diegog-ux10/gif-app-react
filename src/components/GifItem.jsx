import ProTypes from 'prop-types';

export const GifItem = ({title, url}) => {


  return (
    <div className="card">
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
  )
}

GifItem.ProTypes = {
  title: ProTypes.string.isRequired,
  url: ProTypes.string.isRequired
}