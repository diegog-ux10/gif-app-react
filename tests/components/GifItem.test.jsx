import { getByRole, render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Tests in Component GifItem', () => {
    
    const title = 'Messi';
    const url = 'http://messi.com/messi.jpg'

    test('should match snapshot', () => {
        const {container} = render(<GifItem />);
        expect(container).toMatchSnapshot();
    })

    test('should match img and alt', () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug();
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('should exist title', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
});
