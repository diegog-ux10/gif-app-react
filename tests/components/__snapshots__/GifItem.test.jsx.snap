// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Component GifItem should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img />
    <p />
  </div>
</div>
`;
